src/components/VVideoPlayer/index.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+  <video
+    @[mouseoverName]="turnOn"
+    @[mouseleaveName]="tyrnOff"
+    @pause="pause"
+    @ended="pause"
+    @keyup="changeSpeed"
+    ref="video"
+    id="video"
+    :poster="previewImageLink"
+    :controls="!isPreview"
+    :[muted]="isPreview"
+    title="Buttons W and S can change speed of your player"
+  >
+    <source
+      :src="link + '#t=50'"
+      type="application/x-mpegURL"
+    />
+  </video>
+</template>
+
+<script setup lang="ts">
+import { defineProps, onMounted, onUpdated, ref, computed } from 'vue'
+import Hls from 'hls.js'
+
+export interface Props {
+  previewImageLink?: string
+  link?: string
+  isPreview?: boolean
+  progress?: number
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  previewImageLink: '',
+  link: '',
+  isPreview: false,
+  progress: 0
+})
+
+const emit = defineEmits(['pause'])
+
+const video = ref()
```

Comment:
I'm not sure about declaring variables without initializing, `video` remains `underfined`. 
I like writing `const video = ref(null)`

Reply:
I've fixed it, also I am not sure about it

thanks for looking at my code